### PR TITLE
002 view job spec

### PR DESCRIPTION
**Pull request by SeanMcAvoy29**

### CODE REVIEW

File: model/jobSpec.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export class JobSpec{
+    jobRole? : string
```
Comment (glenn-hamilton) [inline model/jobSpec.ts]: Should these be optional?

File: service/jobspecService.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { JobSpec } from '../model/jobSpec';
+const axios = require('axios');
+
+
+module.exports.getJobspecById = async function (id: number): Promise<JobSpec> {
+    try {
+        const response = await axios.get('http://localhost:8080/api/job-specification/'+ id);
```
Comment (glenn-hamilton) [inline service/jobspecService.ts]: The domain of the API shouldn't be hard coded into the application, it should come via an environment variable

File: controller/jobspecController.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { JobSpec } from '../model/jobSpec';
+import { Application, Request, Response } from "express";
+const jobService = require('../service/jobspecService')
+
+module.exports = function(app: Application){
+
+    app.get('/job-spec/:id', async (req: Request, res: Response) =>{
```
Comment (glenn-hamilton) [inline controller/jobspecController.ts]: Should this be `job-spec` or `job-specification` (just for consistency)?

File: controller/jobspecController.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        let data:JobSpec;
+
+        try {
+            data = await jobService.getJobspecById(req.params.id)
+        } catch (e) {
+            console.error(e);
```
Comment (glenn-hamilton) [inline controller/jobspecController.ts]: Would it make sense here that if you can't find a job spec you should show a `404` page to the user?

File: views/view-job-spec.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Specification</title>
```
Comment (glenn-hamilton) [inline views/view-job-spec.html]: Would it make sense to add the Job Specification Name into the title?

File: package.json [hunk 6]
```diff
@@ -5,19 +5,26 @@
...
+    "test-ui": "mocha -require ts-node/register --timeout 10000 test/ui/**/*.ts"
   },
   "author": "",
   "license": "ISC",
   "dependencies": {
     "@types/express-session": "^1.17.7",
     "axios": "^1.5.0",
+    "axios-mock-adapter": "^1.22.0",
+    "chai": "^4.3.8",
+    "dotenv": "^16.3.1",
```
Comment (glenn-hamilton) [inline package.json]: You can use the environment variables without the `dotenv` dependency. If you remove it you can use the environment variables as you have already used them. If adding environment variables could you also add them to a list in the `README.md` file?

File: package.json [hunk 7]
```diff
@@ -5,19 +5,26 @@
...
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "mocha -require ts-node/register test/unit/**/*.ts",
+    "test-ui": "mocha -require ts-node/register --timeout 10000 test/ui/**/*.ts"
   },
   "author": "",
   "license": "ISC",
   "dependencies": {
     "@types/express-session": "^1.17.7",
     "axios": "^1.5.0",
+    "axios-mock-adapter": "^1.22.0",
```
Comment (glenn-hamilton) [inline package.json]: Should this be a normal dependency, or a dev dependency?

File: service/jobspecService.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+module.exports.getJobspecById = async function (id: number): Promise<JobSpec> {
+    if(id == null){
+        throw new Error('Invalid ID')
+    }
+    
+    try {
+        const response = await axios.get(process.env.URL+'/job-specification/'+ id);
+        return response.data;
+    } catch (e) {
+        if(e.response.status === 400){
```
Comment (glenn-hamilton) [inline service/jobspecService.ts]: Is this the correct status code to use when something isn't found?

File: model/jobSpec.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+export class JobSpec{
```
Comment (shaunganley) [inline model/jobSpec.ts:1]: don't you also need a sharepoint link?

**Review by shaunganley — APPROVED** [review]
approving but looks like your missing a sharepoint link and you need to link to your page from the job role table if that exists yet

**Comment by shaunganley** [comment]
also, do you have UI tests yet that need updated?

**Pull request merged**